lib/tasks/va_facilities.rake

```diff
@@ -68,7 +61,228 @@ namespace :va_facilities do
+              facility.street_address_zip_code = vaf["Street_Address_Zip_Code"]
+              facility.street_address_zip_code_extension = vaf["Street_Address_Zip_Code_Extension"]
+              facility.county_street_address = vaf["County_Street_Address"]
+              facility.mailing_address = vaf["Mailing_Address"]
+              facility.mailing_address_city = vaf["Mailing_Address_City"]
+              facility.mailing_address_state = vaf["Mailing_Address_State"]
+              facility.mailing_address_zip_code = vaf["Mailing_Address_Zip_Code"]
+              facility.mailing_address_zip_code_extension = vaf["Mailing_Address_Zip_Code_Extension"]
+              facility.county_mailing_address = vaf["County_Mailing_Address"]
+              facility.station_phone_number = vaf["Station_Phone_Number"]
+              facility.station_main_fax_number = vaf["Station_Main_Fax_Number"]
+
+              #these 7 not in Vet Center schema...
+              if (!vaf["Official_Station_Name"].include? "Vet Center")
+                facility.after_hours_phone_number = vaf["After_Hours_Phone_Number"]
+                facility.pharmacy_phone_number = vaf["Pharmacy_Phone_Number"]
+                facility.enrollment_coordinator_phone_number = vaf["Enrollment_Coordinator_Phone_Number"]
+                facility.patient_advocate_phone_number = vaf["Patient_Advocate_Phone_Number"]
+                facility.va_secretary_10n_approved_date = vaf["VA_Secretary_10N_Approved_Date"]
+                facility.planned_activation_date = vaf["Planned_Activation_Date"]
+                facility.station_number_suffix_reservation_effective_date = vaf["StationNumberSuffixReservationEffectiveDate"]
+              end
+
+
+              facility.latitude = vaf["Latitude"].to_f
+              facility.longitude = vaf["Longitude"].to_f
+              facility.congressional_district = vaf["Congressional_District"]
+              facility.market = vaf["MARKET"]
+              facility.sub_market = vaf["SUBMARKET"]
+              facility.sector = vaf["SECTOR"]
+              facility.fips_code = vaf["FIPS_Code"]
+              facility.rurality = vaf["Rurality_U_Urban_R_Rural_H_Highly_Rural"]
+              facility.monday = vaf["Monday"]
+              facility.tuesday = vaf["Tuesday"]
+              facility.wednesday = vaf["Wednesday"]
+              facility.thursday = vaf["Thursday"]
+              facility.friday = vaf["Friday"]
+              facility.saturday = vaf["Saturday"]
+              facility.sunday = vaf["Sunday"]
+              facility.hours_note = vaf["HoursNote"]
+              facility.save
+              ctr += 1
+              puts "Updated facility: #{vaf['Official_Station_Name']}, #{ctr.to_s}"
+            end
+          end
+        end
+      end
+    puts "All VA facilities have been created or updated in the DB!"
+  end
+
+  def create_vet_center_facility(visn, vaf, hidden)
+    VaFacility.create!(
```

Comment:
for the vet centers, is the only difference between the other facilities (besides the fact that it is a vet center) the `hidden` field being set to `true` or are there other differences? 🤔 Trying to think if we can not repeat some of this. :)

Reply:
I know... I was trying to do that.. bu there are 7 fields that don't exist in the vet center schema.. and can't add logic within the create statements.  i.e., 
# these 4 not in VetCenter json schema...
                  after_hours_phone_number: vaf["After_Hours_Phone_Number"],
                  pharmacy_phone_number: vaf["Pharmacy_Phone_Number"],
                  enrollment_coordinator_phone_number: vaf["Enrollment_Coordinator_Phone_Number"],
                  patient_advocate_phone_number: vaf["Patient_Advocate_Phone_Number"],

Reply:
Actually - you don't even need to do the || nil... :)  if the field doesn't exist the value of vaf["fieldname"] .. is nil... and since we allow nil/null for almost every field in this table... it works.  I just assumed if the field didn't exist... it would throw an error.  So I was able to get rid of that additional method to just create the Vet Centers...